stl/inc/execution

```diff
@@ -1112,7 +1112,7 @@ bool _All_of_family_parallel(_FwdIt _First, const _FwdIt _Last, _Pr _Pred) {
 template <class _ExPo, class _FwdIt, class _Pr, _Enable_if_execution_policy_t<_ExPo> /* = 0 */>
 _NODISCARD bool all_of(_ExPo&&, _FwdIt _First, _FwdIt _Last, _Pr _Pred) noexcept /* terminates */ {
     // test if all elements in [_First, _Last) satisfy _Pred with the indicated execution policy
-    _REQUIRE_PARALLEL_ITERATOR(_FwdIt);
+    _REQUIRE_CONST_PARALLEL_ITERATOR(_FwdIt);
     _Adl_verify_range(_First, _Last);
     auto _UFirst      = _Get_unwrapped(_First);
     const auto _ULast = _Get_unwrapped(_Last);

```

Comment:
Ditto have we audited all the machinery used to implement these parallel algorithms to ensure they are prepared for iterators that model a stronger concept than their `iterator_category` and/or have no `iterator_category`?